language/move-vm/runtime/src/interpreter.rs

```diff
@@ -826,9 +828,25 @@ impl Frame {
                             |acc, v| acc.add(v.size()),
                         );
                         cost_strategy.charge_instr_with_size(Opcodes::PACK_GENERIC, size)?;
+                        let struct_def = resolver.struct_instantiation_at(*si_idx);
+                        let struct_ty = resolver.struct_type_at(struct_def.get_def_idx());
+                        let is_nominal_resource = struct_ty.is_resource;
+
+                        let is_resource = if is_nominal_resource {
+                            true
+                        } else {
+                            let mut is_resource = false;
+                            for ty in struct_def.get_instantiation() {
+                                if resolver.is_resource(&ty.subst(self.ty_args())?)? {
+                                    is_resource = true;
+                                }
+                            }
+                            is_resource
+                        };
+
```

Comment:
@dariorussi FYI I decided to not worry about the Mutex right now so we can get the changes to the values in. The multithreading story is overall complicated and I don't think we need to get it sorted out in this PR.